test/integration/node-specific/abstract_cursor.test.ts

```diff
@@ -1,114 +1,263 @@
+
+      collection = client.db('abstract_cursor_integration').collection('test');
 
-    it('throws when mapping to `null` and cleans up cursor', async function () {
-      const cursor = collection.find();
-      cursor.map(() => null);
+      await collection.insertMany([{ name: 'john doe' }]);
+    });
 
-      const error = await cursor.toArray().catch(e => e);
+    afterEach(async function () {
+      transformSpy.resetHistory();
 
-      expect(error).be.instanceOf(MongoAPIError);
-      expect(cursor.closed).to.be.true;
+      await collection.deleteMany({});
+      await client.close();
     });
-  });
 
-  context('Symbol.asyncIterator() with custom transforms', function () {
-    for (const value of falseyValues) {
-      it(`supports mapping to falsey value '${inspect(value)}'`, async function () {
-        const cursor = collection.find();
-        cursor.map(() => value);
+    describe('tryNext()', function () {
+      context('when there is a transform on the cursor', function () {
+        it('does not transform any documents', async function () {
+          const cursor = collection.find().map(transformSpy);
 
-        let count = 0;
+          await cursor.hasNext();
+          expect(transformSpy.called).to.be.false;
+        });
+      });
+    });
 
-        for await (const document of cursor) {
-          expect(document).to.deep.equal(value);
-          count++;
+    const operations: ReadonlyArray<readonly [string, (arg0: FindCursor) => Promise<unknown>]> = [
+      [
+        'tryNext',
+        (cursor: FindCursor) => {
+          return cursor.tryNext();
+        }
+      ],
+      ['next', (cursor: FindCursor) => cursor.next()],
+      [
+        'Symbol.asyncIterator().next',
+        async (cursor: FindCursor) => {
+          const iterator = cursor[Symbol.asyncIterator]();
+          const doc = await iterator.next();
+          return doc.value;
         }
+      ]
+    ] as const;
 
-        expect(count).to.equal(5);
+    context('when there is a transform on the cursor', function () {
+      for (const [method, func] of operations) {
+        it(`${method}() calls the cursor transform when iterated`, async () => {
+          const cursor = collection.find().map(transformSpy);
+
+          const doc = await func(cursor);
+          expect(transformSpy).to.have.been.calledOnce;
+          expect(doc.name).to.equal('JOHN DOE');
+        });
+
+        it(`when the transform throws, ${method}() propagates the error to the user`, async () => {
+          const cursor = collection.find().map(() => {
+            throw new Error('error thrown in transform');
+          });
+
+          const error = await func(cursor).catch(e => e);
+          expect(error)
+            .to.be.instanceOf(Error)
+            .to.match(/error thrown in transform/);
+          expect(cursor.closed).to.be.true;
+        });
+      }
+
+      it('Cursor.stream() calls the cursor transform when iterated', async function () {
```

Comment:
Same is the above comment, but Github's diffs are making it problematic for the rest... Can we change the test structures to match as an example:

```ts
describe('#methodName", function () {
  context('when some condition exists', function () {
    it('does/returns/throws something', function () {
      // assertion
    });
  });
});
```